application/CMakeLists.txt

```diff
@@ -14,6 +14,7 @@ target_compile_options(${PROJECT_NAME}
         -Wall
         -Werror
         -Wextra
+        -Wswitch-enum
```

Comment:
Would it make sense specify the warnings in to the top level CMakeList instead? Or create a variable and define them only once?